src/coffee/directives/api/utils/_async.coffee

```diff
@@ -149,6 +149,8 @@ angular.module('uiGmapgoogle-maps.directives.api.utils')
     if angular.isArray collection
       array = collection
     else
+      collection = _.pick collection, (val, propName) ->
```

Comment:
This is not exactly ideal as it is O(n) twice. But until I refactor this whole file it's not worth the hassle.